### PR TITLE
Adding tf.linalg.tensor_diag

### PR DESCRIPTION
**Pull request by Patric-Ramz**
#15400

### CODE REVIEW

**Comment by Patric-Ramz** [comment]
Hi,

I am getting this error when I run my test. I am not sure how to fix it. 

" Cannot cast ufunc 'add' output from dtype('float64') to dtype('int32') with casting rule 'same_kind' " 

I am not sure how to adjust my function to fix this error. Please help me.

Thank you. : )

**Comment by Patric-Ramz** [comment]
Hi,

I misnamed the branch. I added the tensor_diag function and not the linalg.diag function.

Regards,
Rele

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @Patric-Ramz 
Requested some changes :slightly_frowning_face:

**Comment by Patric-Ramz** [comment]
Hi,

Thank you for the review, I will make the changes.

**Comment by Patric-Ramz** [comment]
Hello @zaeemansari70 

I made the changes you requested, but I am still encountering this error.

ivy\functional\backends\numpy\experimental\linear_algebra.py:69:

" numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'add' output from dtype('float64') to dtype('int32') with casting rule 'same_kind' "

In the numpy backend 'output_array' and 'diagonal_to_add' seem to have conflicting data types. I am not sure how to fix this error. Please assist me.

Another issue I ran into is the 'bfloat16' data type is supported only when using the XLA JIT compiler, and I am running the tests on a CPU without XLA JIT. I chose to exclude the 'bfloat16'.

Kind regards,

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -253,3 +253,22 @@ def einsum(equation, *inputs, **kwargs):
 @to_ivy_arrays_and_back
 def adjoint(matrix, name=None):
     return ivy.adjoint(matrix)
+
+
+@with_supported_dtypes(
+    {
+        "2.9.0 and below": (
+            "float16",
```
Comment (zaeemansari70) [inline ivy/functional/frontends/tensorflow/linalg.py:261]: This "float16", is not supported by `linalg` in any framework as far as I know.
Kindly make sure you are using this decorator correctly :slightly_smiling_face: 

Also, as far as the casting error goes, can you please dive more deep and let me know why and where that occurs? :slightly_smiling_face:

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Requested some changes :slightly_smiling_face:

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**